### PR TITLE
Add density to numpy

### PR DESCRIPTION
**Pull request by henryiii**
Numpy interface was missing density, now it's easy to calculate, so adding it back in. Also fixed a bug where the requested storage type was being ignored.

### CODE REVIEW

**Comment by HDembinski** [comment]
I am not going to review the numpy interface, so feel free to go ahead with this. If you want to review, please give a shout. I only strongly care about the main interface of boost.histogram.

**Comment by henryiii** [comment]
The current `bh=` shortcut is a bit criptic. Here are possible suggestions:

1. `bh=True, bh_storage=bh.storage.Double`: This is what we have now.
2.  `object=True, storage=bh.storage.Double`: This is more descriptive, but not clear that it's boost-histogram specific. Looks like a numpy option. *Unlikely* to ever conflict.
3. `histogram_object=True, histogram_storage=bh.storage.Double`: This still doesn't make it very clear that this specific to boost-histogram, and a bit wordy.
4. `bh_object=True, bh_storage=bh.storage.Double`: This is fairly clear that it comes from boost histogram, and still clear enough.

I'm leaning toward option 4, I think it's a nice blend of length + intent. Thoughts?

**Comment by henryiii** [comment]
Added option 5: `object=bh.Histogram, storage=bh.storage.Double`.
